### PR TITLE
Get incremental installs to automatically work

### PR DESCRIPTION
**Pull request by MilesCranmer**
@mkitti I wonder if there is some way to do this automatically https://github.com/MilesCranmer/PySR/issues/257

TODO:

- [x] Figure out some way we can unit-test this.

### CODE REVIEW

**Comment by MilesCranmer** [comment]
Could also do this automatically with
```python
subprocess.call(
    ["julia", "-e", 'rm(joinpath(first(DEPOT_PATH), "packages", "PyCall"), recursive=true)']
)
```
followed by re-trying `julia.install`. But I'm not sure if that is potentially dangerous or not. Wdyt @mkitti?

**Comment by mkitti** [comment]
Umm, what's the primary issue. Is this new to Julia 1.9?

**Comment by MilesCranmer** [comment]
This issue has been around for a while. It seems to be related to when you update your Python or Julia version, and try to re-install PySR, some install preferences stored in PyCall are violated.

**Comment by mkitti** [comment]
This an environment issue:

```
ERROR: LoadError: The following package names could not be resolved: * PyCall (not found in project or manifest)
```

It is missing the UUID.

**Comment by MilesCranmer** [comment]
Cool!
Screenshot 2023-07-02 at 8 09 36 PM

Automatically patches that PyCall error 🥳

File: pysr/test/incremental_install_simulator.dockerfile [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+RUN python3 -m pip install .
+RUN python3 -m pysr install
+
+# Change Python version:
+RUN pyenv install 3.10 && pyenv global 3.10 && pyenv uninstall -f 3.9.2
+RUN python3 -m pip install --upgrade pip
+
+# Second install of PySR:
+RUN python3 -m pip install .
+RUN rm -r ~/.julia/environments/pysr-0.14.2
```
Comment (MilesCranmer) [inline pysr/test/incremental_install_simulator.dockerfile:51]: Need to reference latest version instead

**Comment by xukai92** [comment]
Hi Cranmer, I came from the https://github.com/JuliaPy/PyCall.jl/pull/945 (I was looking into a similar issue a while back).
Just wonder if you have checked https://cjdoris.github.io/PythonCall.jl/stable/juliacall/, which is an alternative to PyJulia that seems to have better Python/Julia environment/version support.

Btw, really like your works on neural-symbolic methods for physics.
I have been following them and also did [something related](https://proceedings.neurips.cc/paper/2021/hash/147540e129e096fa91700e9db6588354-Abstract.html)  :)

**Comment by MilesCranmer** [comment]
Thanks and thanks for sending your exciting paper! Very cool stuff.

Yes I'm indeed interested in potentially switching to PythonCall.jl. From what I've seen the installation process is much nicer, and so it would be great to have it as the PySR -> Julia interface. Just need someone to try 🙏 🙏 . The API is pretty similar so it shouldn't be too bad to switch (basically just `jl.eval` -> `jl.seval`; and then removing `julia_helpers.py`). The only downsides are that (1) it's fairly new whereas PyJulia has been around for a while and is "battle-tested", and (2) it doesn't sound like it can eliminate the precompilation issue we've been fighting for a while with static python installations (like conda). I'm not sure if (1) is an issue in practice though because juliacall is more Julia-heavy (rather than the Python-heavy PyJulia), so it's is probably safer to use longterm and has other advantages.

**Pull request merged**